src/index.js

```diff
@@ -33,42 +37,98 @@ function fetchData(searchTerm) {
+        let articleUrl = 'https://en.wikipedia.org/?curid=' + pageID
+
+        let imgUrl = await getArticleImage(articleTitle, pageID)
+        // console.log(imgUrl)
+
+        generateCard(articleTitle, articleText, articleUrl, imgUrl)
     }
 }
 
+async function getArticleImage(title, pageID) {
+    let url = baseUrl + "?origin=*" + `&action=query&format=json&prop=pageimages&titles=${title}&pithumbsize=260`;        // "?origin=*" to resolve cors cross-origin request issues
+    let imgURL
 
-function generateCard(title, text, url) {
+    let res = await fetch(url)
+    let data = await res.json()
+    try {
+        imgURL = data.query.pages[String(pageID)].thumbnail.source
+        // console.log(data.query.pages[String(pageID)].thumbnail.source)
+    } catch (err) {
+        imgURL = '../img/default.jpg'
+    }
+
+    // console.log(imgURL)
+    return imgURL
+}
+
+function prepareResultsSection() {
+    resultsSection.textContent = ''
+    let containerDiv = document.createElement('div')
+    let rowDiv = document.createElement('div')
+    let colDiv = document.createElement('div')
+    let h2 = document.createElement('h2')
+
+    containerDiv.className = "container"
+    rowDiv.className = "row"
+    rowDiv.id = "found"
+    colDiv.className = "col-12"
+    h2.classList.add("mb-3", "text-danger")
+    h2.textContent = "Your results: "
+    colDiv.appendChild(h2)
+    rowDiv.appendChild(colDiv)
+    containerDiv.appendChild(rowDiv)
+    resultsSection.appendChild(containerDiv)
+    /*
+        resultsSection.innerHTML = `
+        <div class="container">
+            <div class="row" id="found">
+                <div class="col-12">
+                    <h2 class="mb-3 text-danger">Your results: </h2>
+                </div>
+            </div>
+        </div>
+        `
+    */
+}
+
+
+function generateCard(title, text, url, img) {
+    let foundDiv = document.getElementById('found')
     // console.log(title, text, url)
 
     if (!title || !text || !url) {
         foundDiv.textContent = 'No results found'
     } else {
         foundDiv.innerHTML += `
-        <div class="col-md-6 col-lg-4">
+        <div class="col-sm-12 col-md-6 col-lg-4" >
             <div class="card my-3">
                 <div class="card-thumbnail">
-                    <img src="./img/default.jpg" class="img-fluid" alt="thumbnail">
+                    <img src="${img}" class="img-fluid" alt="${title}">
                 </div>
                 <div class="card-body">
-                    <h3 class="card-title"><a href="#" class="text-secondary">${title}</a></h3>
+                    <h3 class="card-title fs-4">${title}</h3>
                     <p class="card-text">${text}</p>
-                    <a href="${url}" target="blank" class="btn btn-danger">Read More</a>
+                    <a href="${url}" target="blank" class="btn btn-info link-btn">Read More</a>
                 </div>
             </div>
-        </div>
-        
+        </div >
         `
```

Comment:
Boy oh boy, this is how you use innerHTML!
Saved my time looking for how to code raw HTML in JS.

btw do you think I can use ${} for the attributes we wanna add?
e.g. let gootstrap_gridSys = "col-sm-12 col-md-6 col-lg-4"
`<div class=${bootstrap_gridSys} >`

Can't wait to experiment! Thanks for the tips!